ui/apps/platform/src/hooks/useURLSort.ts

```diff
@@ -45,65 +34,161 @@ function tableSortOption(
+    sortOption: QueryValue,
+    defaultSortOption: SortOption
+): SortOption | SortOption[] {
+    if (Array.isArray(sortOption)) {
+        return sortOption.filter(isSortOption);
+    }
+
+    return isSortOption(sortOption) ? sortOption : defaultSortOption;
 }
 
+export type UseURLSortProps = {
+    sortFields: (string | string[])[];
+    defaultSortOption: SortOption;
+    onSort?: (newSortOption: SortOption | SortOption[]) => void;
+};
+
+export type UseURLSortResult = {
+    sortOption: ApiSortOption;
+    setSortOption: (newSortOption: SortOption, historyAction?: HistoryAction | undefined) => void;
+    getSortParams: GetSortParams;
+};
+
+/**
+ * A hook that manages the sort option for a table in the URL.
+ *
+ * @param options.sortFields
+ *      An array of (string | string[]) that represent the sort fields that will
+ *      sent over the API for a given table column. A `string` value represents a
+ *      single field, while a `string[]` value represents multiple fields that may
+ *      have a subset of values sent over the API.
+ * @param options.defaultSortOption
+ *      The default sort option to use when the sort option is not present in the URL.
+ * @param options.onSort
+ *      A callback function that is called when the sort option is changed.
+ * @returns UseURLSortResult.sortOption
+ *      The current sort option. This may be a single
+ *      sort option or an array of sort options.
+ * @returns UseURLSortResult.setSortOption
+ *     A function that can be used to directly set the sort option.
+ * @returns UseURLSortResult.getSortParams
+ *     A function that can be used to get the sort parameters for a given column in a PatternFly table.
+ *     This function has two main signatures for the singular and multi sort use cases:
+ *
+ *     1. getSortParams(columnName: string, fieldOptions?: SortAggregate)
+ *     - This signature is used when the column has a single sort field. The `columnName` parameter
+ *     is used to find the correct column index in the PatternFly table, and is also used as the default
+ *     field name when sending the sort option to the API. The `fieldOptions` parameter is optional and
+ *     is used to specify the aggregate function to use when sorting the column.
+ *
+ *     2. getSortParams(columnName: string, fieldOptions: { field: string; aggregateBy?: SortAggregate }[])
+ *     - This signature is used when the column has multiple sort fields. In this form, the fields sent to
+ *     the API will be the `field` values in the `fieldOptions` array, each of which can optionally
+ *     include an aggregate. The correct column index in the PatternFly table is found by finding any array
+ *     provided in the `sortFields` option that contains one of the `field` values in the `fieldOptions` array.
+ *
+ * @example
+ * // A use case where a table only has single sort columns
+ * const { sortOption, setSortOption, getSortParams } = useURLSort({
+ *    sortFields: ['name', 'age', 'date'],
```

Comment:
Maybe I got lost. Do these correspond to backend sort fields? If yes, is the convention Title Case?

Reply:
They do not, it was a simple example for illustrative purposes. I can change to actual sort fields though to more closely resemble what the API expects.